### PR TITLE
Add more logging. Fix ScriptsClassLoader

### PR DESCRIPTION
**Pull request by gallyamb**

### CODE REVIEW

**Comment by MentionBot-JSFlight** [comment]
@Gebon, thanks for your PR! By analyzing the history of the files in this pull request, we identified @d0k1 to be a potential reviewer.

**Comment by gallyamb** [comment]
Closes #115

File: jmeter/src/main/java/com/focusit/jsflight/jmeter/JMeterScriptProcessor.java [hunk 1]
```diff
@@ -117,6 +117,11 @@ public void processScenario(HTTPSamplerBase sampler, HashTree tree, Arguments us
             return;
         }
         compiledProcessScript.setBinding(binding);
-        compiledProcessScript.run();
+        LOG.info("Run compiled script");
+        try {
+            compiledProcessScript.run();
+        } catch (Throwable throwable) {
+            LOG.error(throwable.getMessage(), throwable);
+        }
```
Comment (gallyamb) [inline jmeter/src/main/java/com/focusit/jsflight/jmeter/JMeterScriptProcessor.java:125]: Нужно сделать такую же штуку для stepProcessor

File: player/src/main/java/com/focusit/jsflight/player/configurations/CommonConfiguration.java [hunk 2]
```diff
@@ -241,17 +239,16 @@ public void setFormOrDialogXpath(String formOrDialogXpath)
...
         scriptClassloaderLock.lock();
         try
         {
             if (scriptClassloader == null)
             {
-                ArrayList<URL> urls = new ArrayList<>();
-                urls.addAll(findClasspathForScripts(System.getProperty("cp")));
-                scriptClassloader = new ScriptsClassLoader(this.getClass().getClassLoader(),
-                        urls.toArray(new URL[urls.size()]));
+                URL[] urls = findClasspathForScripts(System.getProperty("cp")).stream().toArray(URL[]::new);
```
Comment (gallyamb) [inline player/src/main/java/com/focusit/jsflight/player/configurations/CommonConfiguration.java]: Переделать так, чтобы `findClasspathForScripts` сразу возвращала массив

File: script/src/main/java/com/focusit/jsflight/script/ScriptsClassLoaderFactory.java [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+            return null;
+        }
+    }
+
+    private static Class<?>[] convertToPrimitiveClasses(Class[] args)
+    {
+        return Arrays.stream(args).map(arg -> TO_PRIMITIVES.getOrDefault(arg, arg)).toArray(Class<?>[]::new);
+    }
+
+    private static Method getMethodUsingObjectClasses(Class<?> clazz, String name, Class[] args)
```
Comment (gallyamb) [inline script/src/main/java/com/focusit/jsflight/script/ScriptsClassLoaderFactory.java]: Merge with `getMethodUsingPrimitiveClasses`

File: script/src/main/java/com/focusit/jsflight/script/ScriptsClassLoaderFactory.java [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+            }
+            return result;
+        }
+        finally
+        {
+            urlMethod.setAccessible(isUrlMethodAccessible);
+        }
+    }
+
+    private static Object invokeParentClassLoaderMethod(ClassLoader parent, String methodName, Object[] args,
```
Comment (gallyamb) [inline script/src/main/java/com/focusit/jsflight/script/ScriptsClassLoaderFactory.java]: Merge with `invokeUrlClassLoaderMethod`

File: player/src/main/java/com/focusit/jsflight/player/scenario/ScenarioProcessor.java [hunk 5]
```diff
@@ -301,13 +300,18 @@ public void play(UserScenario scenario, SeleniumDriver seleniumDriver, int start
...
-            applyStep(scenario, seleniumDriver, scenario.getPosition());
-            scenario.moveToNextStep();
+        try {
+            while (scenario.getPosition() != maxPosition) {
+                LOG.info("Step " + scenario.getPosition());
+                applyStep(scenario, seleniumDriver, scenario.getPosition());
+                scenario.moveToNextStep();
+            }
+            LOG.info(String.format("Done(%d):playing", System.currentTimeMillis() - begin));
+        } catch (Throwable throwable) {
```
Comment (d0k1) [inline player/src/main/java/com/focusit/jsflight/player/scenario/ScenarioProcessor.java:310]: What is the reason to catch everything?

File: player/src/main/java/com/focusit/jsflight/player/scenario/ScenarioProcessor.java [hunk 6]
```diff
@@ -301,13 +300,18 @@ public void play(UserScenario scenario, SeleniumDriver seleniumDriver, int start
...
+            LOG.error("Playing interrupted by unhandled exception", throwable);
+            throw throwable;
+        } finally {
+            seleniumDriver.closeWebDrivers();
```
Comment (d0k1) [inline player/src/main/java/com/focusit/jsflight/player/scenario/ScenarioProcessor.java:314]: As far as I remember, I wanted to leave browsers opened to have an opportunity to continue playing scenario after some problems gets resolved. 
Imagine a scenario passed 950 step of 1000 and the something came up. 
You could fix some issues and you might want to continue playing the scenario started at 950 step of 1000.

File: script/src/main/java/com/focusit/jsflight/script/ScriptsClassLoaderFactory.java [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+        {
+            return clazz.getDeclaredMethod(name, args);
+        }
+        catch (NoSuchMethodException e)
+        {
+            return null;
+        }
+    }
+
+    public static ClassLoader createScriptsClassLoader(ClassLoader parent, URL... urls)
```
Comment (d0k1) [inline script/src/main/java/com/focusit/jsflight/script/ScriptsClassLoaderFactory.java]: need to discuss this

**Pull request merged**